### PR TITLE
Fix OSError(30) in Calibration, Add Calibration Workflow, Integrate SWxSOC Package, and Fix Linting Issues

### PR DESCRIPTION
**Pull request by dbarrous**
This pull request includes the following changes and improvements:

- **Fixes OSError(30) in Lambda Function:** Modifies the calibration function to write new files to a temporary directory. This prevents the OSError(30, 'Read-only file system') error when performing HDUList.writeto in the Lambda function.

- **Adds Calibration Workflow:** Implements a new calibration workflow that runs the calibration process and comments with a link to the processed artifacts. Adds a test event for the workflow to utilize. Includes a correctly named test bin file for the workflow.

- **Integrates SWxSOC Core Package:** Adds the SWxSOC core package as a dependency for creating and parsing filenames. Removes unused duplicate functions and tests for `utils`, updating the tests and code to use functions from the SWxSOC package.

- **Fixes Linting Issues:** Resolves various linting issues across the codebase.

### CODE REVIEW

**Comment by dbarrous** [comment]
This fix is able to produce the file `padre_meddea_l0test_eventlist_20240327T134616_v0.1.0.fits`, @ehsteve Is it supposed to be `l0` or `l1`? I can change it back to the latter if needed.

**Comment by dbarrous** [comment]
Since this isn't a `pull_request_target` event with proper write permissions since the workflow is being added in `this` PR, it won't comment. But will comment with artifact link on consequent PRs. Here is the link of the generated artifacts: https://github.com/PADRESat/padre_meddea/actions/runs/10177912394/artifacts/1759407468

File: padre_meddea/tests/test_data/test_meddea_event.json
Comment (Alrobbertz) [inline padre_meddea/tests/test_data/test_meddea_event.json:1]: Is this file needed here as well as in the workflows/test_data?

File: padre_meddea/util/config.py
Comment (Alrobbertz) [inline padre_meddea/util/config.py:1]: Can all this be removed if we're using the `swxsoc.config` ?

File: padre_meddea/util/util.py [hunk 1]
```diff
@@ -2,183 +2,10 @@
...
-
-
-__all__ = ["create_science_filename", "parse_science_filename", "VALID_DATA_LEVELS"]
+__all__ = ["VALID_DATA_LEVELS"]
 
 TIME_FORMAT_L0 = "%Y%j-%H%M%S"
 TIME_FORMAT = "%Y%m%dT%H%M%S"
 VALID_DESCRIPTORS = ["eventlist", "spec-eventlist", "spec", "xraydirect"]
 VALID_DATA_LEVELS = ["l0", "l1", "l2", "l3", "l4"]
 FILENAME_EXTENSION = ".fits"
```
Comment (Alrobbertz) [inline padre_meddea/util/util.py:11]: Can these also be removed if we're using the swxsoc config?

Comment (dbarrous) [reply padre_meddea/util/util.py]: Yeah good catch on all of these!

File: .gitignore [hunk 2]
```diff
@@ -181,7 +181,7 @@ Temporary Items
 # jupyter notebooks should not be added to Python Packages
 *.ipynb
 
-*.bin
+# *.bin
```
Comment (Alrobbertz) [inline .gitignore:184]: Can you uncomment this?

File: padre_meddea/tests/test_util_util.py
Comment (Alrobbertz) [inline padre_meddea/tests/test_util_util.py:1]: Can this and the `util.py` be completely removed? It looks like you have almost all of the content removed from here anyway besides some import statements.

File: padre_meddea/util/config.py
Comment (Alrobbertz) [inline padre_meddea/util/config.py:1]: can this whole file be removed?

File: padre_meddea/util/util.py
Comment (Alrobbertz) [inline padre_meddea/util/util.py:1]: Same here - it looks like you remove all the meaningful content besides some import statements. Can we remove the while file?

**Review by Alrobbertz — APPROVED** [review]
LGTM!
